packages/components/src/geolocation/GeolocationProvider.tsx

```diff
@@ -0,0 +1,112 @@
+import { useTranslation } from 'next-i18next';
+import { createContext, useContext, useState, useCallback } from 'react';
+import type { ReactNode, Context } from 'react';
+import type { Coordinates } from '../types';
+import sendNotification from '../utils/sendToast';
+import geolocationService from './GeolocationService';
+
+export type GeolocationContextType = {
+  coordinates: Coordinates | null;
+  loading: boolean;
+  called: boolean;
+  error?: GeolocationPositionError | null;
+  resolve: () => Promise<Coordinates | null>;
+};
+
+const GeolocationContext: Context<GeolocationContextType> =
+  createContext<GeolocationContextType>({
+    coordinates: null,
+    loading: false,
+    called: false,
+    error: null,
+    resolve: () => Promise.resolve(null),
+  });
+
+type Props = {
+  children: ReactNode;
+};
+
+export const isGeolocationPositionError = (
+  error: unknown
+): error is GeolocationPositionError =>
+  Boolean(
+    error &&
+      typeof error === 'object' &&
+      'code' in error &&
+      'message' in error &&
+      'PERMISSION_DENIED' in error &&
+      'POSITION_UNAVAILABLE' in error &&
+      'TIMEOUT' in error
+  );
+
+export default function GeolocationProvider({ children }: Props) {
```

Comment:
Imho Context+provider is needed when some state is needed globally + can be read but also _changed_ anywhere in the sub tree components. I mean, the user does somehting in the ui that changes the global state. But with location it's not possible? So I would think that we dont need contextProvider, but only a hook like `useGeolocation` which would only return the location values and handle the error notification.

Reply:
Something like this: https://itnext.io/creating-react-useposition-hook-for-getting-browsers-geolocation-2f27fc1d96de

Reply:
As discussed, this is refactoring which probably is not the most important thing right now, so let's not do it especially because the functionality works correctly now.